docs/architecture/adr-011-light-client-crates-extraction.md

```diff
@@ -0,0 +1,292 @@
+# ADR 011: Light client crates extraction
+
+## Changelog
+
+* 2022-07-29: Initial draft
+
+## Context
+
+The current design of the `ibc` (modules) crate precludes light client implementations from being hosted outside of the
+ibc-rs repository. This is primarily due to circular dependencies and tight coupling between the light client
+implementations and the `ibc` crate.
+
+For e.g., the Tendermint light client implementation (rightly) depends on the `ibc` crate for the trait definitions
+(such as `ClientState`) that a light client is expected to implement, but the `ibc` crate also depends on the Tendermint
+light client's `ClientState` impl to define its `AnyClientState` enum:
+
+```rust
+pub enum AnyClientState {
+    Tendermint(client_state::ClientState),
+
+    #[cfg(any(test, feature = "mocks"))]
+    Mock(MockClientState),
+}
+```
+
+See [ADR003 - Dealing with chain-specific datatypes](https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-003-handler-implementation.md#dealing-with-chain-specific-datatypes)
+for the rationale behind the `Any*` enum design choice.
+
+Additionally, there are places where the core modules code (indirectly) depends on light client specific types. e.g.:
```

Comment:
Pointing out the exact dependencies in the code snippet would be helpful, something like:
```suggestion
Additionally, there are places where the core modules code (indirectly) depends on light client specific types, such as the `tendermint::abci::Event` and `IbcEvent` types in the following code snippet:
```

Reply:
Fixed in https://github.com/informalsystems/ibc-rs/pull/2356/commits/5d2495c3e645ca215f9c76f34eff2080c2d0283d